v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/CsvConverters.java

```diff
@@ -0,0 +1,540 @@
+import org.apache.beam.sdk.transforms.View;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.apache.beam.sdk.values.PCollectionView;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.sdk.values.TupleTagList;
+import org.apache.beam.vendor.guava.v20_0.com.google.common.base.Throwables;
+import org.apache.beam.vendor.guava.v20_0.com.google.common.io.CharStreams;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+
+/** Common transforms for Csv files. */
+public class CsvConverters {
+
+  private static final String SUCCESSFUL_TO_JSON_COUNTER = "SuccessfulToJsonCounter";
+
+  private static final String FAILED_TO_JSON_COUNTER = "FailedToJsonCounter";
+
+  /**
+   * The {@link CsvConverters#getSchema(String)} gets a json schema from GCS and returns it as a
+   * non-parsed json string.
+   *
+   * @param jsonSchemaPath path to json schema
+   * @return non-parsed json schema string
+   * @throws IOException thrown if not able to read schema
+   */
+  private static String getSchema(String jsonSchemaPath) {
+    MatchResult result;
+    try {
+      result = FileSystems.match(jsonSchemaPath);
+      checkArgument(
+          result.status() == MatchResult.Status.OK && !result.metadata().isEmpty(),
+          "Failed to match any files with the pattern: " + jsonSchemaPath);
+
+      return result.metadata().stream()
+          .map(MatchResult.Metadata::resourceId)
+          .map(
+              resourceId -> {
+                try (Reader reader =
+                    Channels.newReader(
+                        FileSystems.open(resourceId), StandardCharsets.UTF_8.name())) {
+                  return CharStreams.toString(reader);
+                } catch (IOException e) {
+                  throw new UncheckedIOException(e);
+                }
+              })
+          .collect(Collectors.joining());
+    } catch (IOException ioe) {
+      throw new RuntimeException("File system i/o error: " + ioe.getMessage());
+    }
+  }
+
+  /**
+   * Builds Json string from list of values and headers or values and schema if schema is provided.
+   *
+   * @param headers optional list of strings which is the header of the Csv file.
+   * @param values list of strings which are combined with header or json schema to create Json
+   *     string.
+   * @param jsonSchemaString
+   * @return Json string containing object.
+   * @throws IOException thrown if Json object is not able to be written.
+   */
+  static String buildJsonString(
+      @Nullable List<String> headers, List<String> values, @Nullable String jsonSchemaString)
+      throws IOException {
+    StringWriter stringWriter = new StringWriter();
+    JsonWriter writer = new JsonWriter(stringWriter);
+    if (jsonSchemaString != null) {
+      JsonArray jsonSchema = new JsonParser().parse(jsonSchemaString).getAsJsonArray();
```

Comment:
Can a JsonParser be reused? If so, better to create a static parser instead of creating a parser for every record

Reply:
Yes static parser has been created and will be reused.